src/main/java/org/openstreetmap/atlas/generator/tools/spark/SparkJob.java

```diff
@@ -425,8 +426,9 @@ private void writeStatus(final String path, final String name, final String cont
         try
         {
             final FileSystem fileSystem = getFileSystem(path);
-            final BufferedWriter out = new BufferedWriter(
-                    new OutputStreamWriter(fileSystem.create(new Path(path + "/" + name))));
+            final BufferedWriter out = new BufferedWriter(new OutputStreamWriter(
+                    fileSystem.create(new Path(SparkFileHelper.combine(path, name)))));
+            // TODO this leaks a resource if write() throws an exception
```

Comment:
Can you put the `BufferedWriter` in a try-with-resources block to avoid the leak?

Reply:
I don't see why not.